test/tests/functional/pbs_eligible_time.py

```diff
@@ -84,27 +86,45 @@ def test_job_array(self):
 
         J1 = Job(TEST_USER, attrs={ATTR_J: '1-3'})
         J1.set_sleep_time(20)
+        now = int(time.time())
         jid = self.server.submit(J1)
         jid_short = jid.split('[')[0]
         sjid1 = jid_short + '[1]'
         sjid2 = jid_short + '[2]'
         sjid3 = jid_short + '[3]'
 
+        # The job array has [] in its ID.
+        # These are regexp, so they need to be escaped.
+        jid_esp = jid.replace('[', '\[').replace(']', '\]')
+
+        # Capture the time stamp when accrue type changed to
+        # eligible time
+        msg1 = jid_esp + ";Accrue type has changed to eligible_time"
+        m1 = self.server.log_match(msg1, starttime=now, regexp=True)
```

Comment:
to be on safer side use now-1
Also there is no regular expression in your log match. you should remove it.

Reply:
@anamikau , the regular expression is still required to match the jobid with [ and ].
Also, the 'now' time is even before the job starts. So I feel that would be safer.

Reply:
No, you shouldn't need regex. The string will end up as: `23[1].server;Accrue type...`, which can be done without regex.

You would also have to remove the escaping of `[]`